WordPress/Classes/Networking/SharingServiceRemote.swift

```diff
@@ -290,6 +292,97 @@ public class SharingServiceRemote : ServiceRemoteREST
+
+                onSuccess(sharingButtons)
+            },
+            failure: { (operation: AFHTTPRequestOperation?, error: NSError) in
+                failure?(error)
+        })
+    }
+
+
+    /// Updates the list of sharing buttons for a blog.
+    ///
+    /// - Parameters:
+    ///     - siteID: The WordPress.com ID of the site.
+    ///     - sharingButtons: The list of sharing buttons to update. Should be the full list and in the desired order.
+    ///     - success: An optional success block accepting an array of `RemoteSharingButton` objects.
+    ///     - failure: An optional failure block accepting an `NSError` argument.
+    ///
+    public func updateSharingButtonsForSite(siteID: NSNumber, sharingButtons:[RemoteSharingButton], success: (([RemoteSharingButton]) -> Void)?, failure: (NSError! -> Void)?) {
+        let endpoint = "sites/\(siteID)/sharing-buttons"
+        let path = pathForEndpoint(endpoint, withVersion: ServiceRemoteRESTApiVersion_1_1)
+        let parameters = [SharingButtonsKeys.sharingButtons : sharingButtons]
+
+        api.POST(path,
+            parameters: parameters,
+            success: { (operation: AFHTTPRequestOperation!, response: AnyObject!) in
+                guard let onSuccess = success else {
+                    return
+                }
+
+                let responseDict = response as! NSDictionary
+                let buttons = responseDict.arrayForKey(SharingButtonsKeys.updated)
+                let sharingButtons = self.remoteSharingButtonsFromDictionary(buttons)
+
+                onSuccess(sharingButtons)
+            },
+            failure: { (operation: AFHTTPRequestOperation?, error: NSError) in
+                failure?(error)
+        })
+    }
+
+
+    /// Composees a `RemotePublicizeConnection` populated with values from the passed `NSDictionary`
+    ///
+    /// - Parameters:
+    ///     - buttons: An `NSArray` of `NSDictionary`s representing `RemoteSharingButton` objects.
+    ///
+    /// - Returns: An array of `RemoteSharingButton` objects.
+    ///
+    private func remoteSharingButtonsFromDictionary(buttons: NSArray) -> [RemoteSharingButton] {
+        var order = 0;
+        let sharingButtons: [RemoteSharingButton] = buttons.map { (let dict) -> RemoteSharingButton in
+            let btn = RemoteSharingButton()
+            btn.buttonID = dict.stringForKey(SharingButtonsKeys.buttonID) ?? btn.buttonID
+            btn.name = dict.stringForKey(SharingButtonsKeys.name) ?? btn.name
+            btn.shortname = dict.stringForKey(SharingButtonsKeys.shortname) ?? btn.shortname
+            btn.custom = dict.numberForKey(SharingButtonsKeys.custom).boolValue ?? btn.custom
+            btn.enabled = dict.numberForKey(SharingButtonsKeys.enabled).boolValue ?? btn.enabled
+            btn.visibility = dict.stringForKey(SharingButtonsKeys.visibility) ?? btn.visibility
+            btn.order = order
+            order++
```

Comment:
`++` operator will be deprecated in Swift 2.2, and removed from Swift 3: https://github.com/apple/swift-evolution/blob/master/proposals/0004-remove-pre-post-inc-decrement.md